File: Algorithms/src/andy/javascript/level1/RecommendNewId.js
// 신규 아이디 추천
function solution(new_id) {
    var answer = '';
    new_id = new_id.toLowerCase();
    let str = "";
    [...new_id].forEach( (d) => {
    	if( (d >='a' && d<='z') || (d >='0' && d<='9') || d == "-" || d == "_" || d == ".") {
    		str += d;
    	}
    });
    str = str.replace(/\.{1,}/gi,".");
    if(str.charAt(0) == ".") str = str.substr(1);
    if(str.charAt(str.length-1) == ".") str = str.substr(0,str.length-1);
    if(str.length == 0) str = "a";
    if(str.length>15) str = str.substr(0,15);
    if(str.charAt(str.length-1) == ".") str = str.substr(0,str.length-1);
    if(str.length<=2) str = str + str.charAt(str.length-1).repeat(3-str.length);
    answer = str;
    return answer;
}

File: Andy/src/level1/java/RecommendNewId.java
package level1.java;

// 신규 아이디 추천
public class RecommendNewId {
    public String solution(String new_id) {
        new_id = new_id.toLowerCase();
        new_id = new_id.replaceAll("[^a-z0-9._-]", "");
        new_id = new_id.replaceAll("[.]{1,}", ".");
        if(new_id.length()>0 && new_id.charAt(0) == '.') new_id = new_id.substring(1);
        if(new_id.length()>0 && new_id.charAt(new_id.length()-1) == '.') new_id = new_id.substring(0,new_id.length()-1);
        if(new_id.length()==0) new_id = "a";
        if(new_id.length()>15) new_id = new_id.substring(0,15);
        if(new_id.charAt(new_id.length()-1) == '.') new_id = new_id.substring(0,new_id.length()-1);
        if(new_id.length() <= 2) {
        	String tmp = ""+ new_id.charAt(new_id.length() - 1);
        	new_id = new_id + tmp.repeat(3-new_id.length());
        }
        return new_id;
    }	

}
